interview-questions.md

```diff
@@ -10,46 +10,83 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
 Your answer:
 
-Researched answer:
+Yarn is a package manager for Jest. It holds all of the directories and built in functions needed to run your Jest tests for JavaScript. Once you get "good failure" on your test, function not defined, you can write your function and play around until the function until it passes the test you coded.
+
+When you run yarn in your terminal it will modify all of the files inside the repo your branch is in. Or if you mess up, like I did, and run yarn when you are on your desktop it will modify all of the folders that hold Jest test on your desktop...
+   
+
+Researched answer: Resource (https://flaviocopes.com/yarn/)
+
+Created in 2016, yarn is one of the main JavaScript package managers for node.js. Yarn has the ability to install packages from your local cache. When you initiate a new project, through your terminal, yarn writes its dependencies into a file marked package.json. It is stored in the root folder of your project, and stores the dependencies files that effect your entire project.
 
 3. What is a React component?
 
 Your answer:
 
-Researched answer:
+React components are the external files, or the children, that are seperated from your main App.js and they communicate with your app through component calls at the bottom of your App.js (ex: <Button />). You can store your code for a specific item outside your code to keep it clean and compartmentalized.
+
+Researched answer: Resource (https://www.w3schools.com/react/react_components.asp#:~:text=Components%20are%20independent%20and%20reusable,will%20concentrate%20on%20Function%20components.)
+
+React components are small bits of code that can be reused independently. They are similar to a JavaScript function, but work independently and return HTML. There are two types of React components. First there are Class components. These must have an "extends React.Component statement to inherite access to React.Component's functions. They also must have a render() method, so that they can return HTML.Then there are Functional components. Function components return HTML as well, but can be written with less code, in a cleaner way which increases the ease of use and understanding. With React components you can pass them throughout your project as properties, or Props for short. Think of a Prop as a function argument that can be passed to components as attributes.
```

Comment:
Love that you mentioned functional and class components because while one may be more popular, the other still exists and you may come across it.